lib/pat.c

```diff
@@ -1215,6 +1215,16 @@ pat_get(grn_ctx *ctx, grn_pat *pat, grn_id id)
   return grn_io_array_at(ctx, pat->io, SEGMENT_PAT, id, &flags);
 }
 
+static inline pat_node_common *
+_pat_get(grn_ctx *ctx, grn_pat *pat, grn_id id)
+{
+  int flags = GRN_TABLE_ADD;
+  if (id > GRN_ID_MAX) {
+    return NULL;
+  }
```

Comment:
We could skip the unnecessary assignment.
```suggestion
  if (id > GRN_ID_MAX) {
    return NULL;
  }
  int flags = GRN_TABLE_ADD;
```

Reply:
I've fixed it.